File: solutions/ruby/resistor-color-trio/1/resistor_color_trio.rb
class ResistorColorTrio
  COLORS = %w[black brown red orange yellow green blue violet grey white]
  
  def initialize(colors)
    @colors = colors
    calc
  end

  def calc 
    @value = ((COLORS.index(@colors[0]) * 10) + COLORS.index(@colors[1])) * 10 ** COLORS.index(@colors[2])

    if @value % 1000000000 == 0 &&  @value.nonzero?
      @unit = 'gigaohms'
      @value /= 1000000000
    elsif @value % 1000000 == 0 &&  @value.nonzero?
      @unit = 'megaohms'
      @value /= 1000000
    elsif @value % 1000 == 0 &&  @value.nonzero?
      @unit = 'kiloohms'
      @value /= 1000
    else
      @unit = 'ohms'
    end
  end
  
  def label
    "Resistor value: #{@value} #{@unit}"
  end
end

File: solutions/java/resistor-color-trio/1/src/main/java/ResistorColorTrio.java
import java.util.List;

class ResistorColorTrio {
    private final List<String> r = List.of(
        "black","brown","red","orange","yellow",
        "green","blue","violet","grey","white"
    );

    String label(String[] colors) {
        StringBuilder units = new StringBuilder();

        long value = (r.indexOf(colors[0]) * 10 + r.indexOf(colors[1])) * (long)Math.pow(10, r.indexOf(colors[2]));

        if (value % 1_000_000_000 == 0 && value != 0) {
            value /= 1_000_000_000;
            units.append(" gigaohms");
        } else if (value % 1_000_000 == 0 && value != 0) {
            value /= 1_000_000;
            units.append(" megaohms");
        } else if (value % 1_000 == 0 && value != 0) {
            value /= 1_000;
            units.append(" kiloohms");
        } else {
            units.append(" ohms");
        }

        return value + units.toString();
    }
}